flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/InternalSourceReaderMetricGroup.java

```diff
@@ -144,11 +144,20 @@ public void updateMaxDesiredWatermark(long currentMaxDesiredWatermark) {
     }
 
     boolean isIdling() {
+        return isIdling(idleStartTime);
+    }
+
+    private boolean isIdling(long idleStartTime) {
         return idleStartTime != ACTIVE;
     }
 
     long getIdleTime() {
-        return isIdling() ? this.clock.absoluteTimeMillis() - idleStartTime : 0;
+        // this class is not thread-safe, use the local variable to get a snapshot value.
+        long currentIdleStartTime = idleStartTime;
```

Comment:
I am curious about whether you think we need the high number set at all. Could we set to 0 for active in line with the way the getter returns?  

The code would be simpler if we can remove the internal high number.